### DISCUSSION TITLE
@cypress/xpath no longer supported?

### DISCUSSION THREAD

**Discussion by jgonzalezalbisu** [cypress-io/cypress#27497] [Questions and Help]
I've seen a recent message pop up when installing dependencies that `@cypress/xpath` (https://www.npmjs.com/package/@cypress/xpath) is no longer supported and deprecated? Is there an alternative for this plugin?

**Comment by portyu9** [comment]
Hi @jgonzalezalbisu 

Yes. The Cypress XPath package has been deprecated and is no longer supported.

For new or migrated tests, prefer Cypress's native querying APIs:

- `cy.get()` with CSS selectors
- `cy.contains()` when visible text is the meaningful selector
- stable testing attributes such as `data-cy`

For example, instead of an XPath tied to page structure:

```js
cy.xpath('//button[@id="submit"]')
```

prefer something like:

```js
cy.get('[data-cy="submit"]')
```

Cypress specifically recommends dedicated `data-*` attributes because they are less coupled to CSS styling and application behavior.

If you already have a large XPath suite, I would migrate the XPath expressions incrementally rather than adding a new dependency on an unsupported plugin.

Please mark this answer as accepted if it helped, thank you.
